packages/playback/test/lib/pipelines/native/native-pipeline.test.ts

```diff
@@ -0,0 +1,261 @@
+import type {
+  AudioTrack,
+  AudioTrackList,
+  IPipelineDependencies,
+  IRemoteVttThumbnailTrackOptions,
+} from 'src/entry-points/api-reference';
+import { NativePipeline } from '../../../../src/lib/pipelines/native/native-pipeline';
+import { beforeEach, describe, expect, it } from 'vitest';
+import { TextTrackKind, TextTrackMode } from '../../../../src/lib/consts/text-tracks';
+import { PlaybackState } from '../../../../src/lib/consts/playback-state';
+
+describe('NativePipeline', () => {
+  let nativePipeline: NativePipeline;
+  let videoElement: HTMLVideoElement;
+  beforeEach(() => {
+    videoElement = document.createElement('video');
+    nativePipeline = NativePipeline.create({
+      logger: {
+        createSubLogger: (name) => {
+          name;
+        },
+      },
+      videoElement,
+      source: {
+        url: new URL('https://foo.bar.m3u8'),
+      },
+    } as IPipelineDependencies);
+
+    // text mocks.
+    window.HTMLMediaElement.prototype.addTextTrack = (kind: string, label?: string, language?: string): TextTrack => {
+      const textTrack = {
+        kind,
+        label,
+        language,
+        mode: TextTrackMode.Hidden,
+        id: `id-${label}-${language}`,
+      } as TextTrack;
+      videoElement.textTracks[videoElement.textTracks.length] = textTrack;
+      return textTrack;
+    };
```

Comment:
you can use mockito, or simply `videoElement = { addTextTrack: vi.fn() }` in beforeEach

Reply:
The only issue I have with mocking this with an empty vi.fn is that it doubles as a test utility. If this is mocked with vi, the text track would need to be added manually in each test.

Reply:
I think this is fine

Reply:
Cool, fixing the size limits and this should be good to go.